### PR TITLE
Fix flaky backlight test

### LINKED ISSUE #653: Fix unstable test for grid_detect_then_xray_centre_plan
Our unit test `test_detect_grid_and_do_gridscan` occasionally fails. This is likely due to something not being cleaned up properly. See https://github.com/DiamondLightSource/mx-bluesky/actions/runs/11859361401/job/33052132442 for an instance of it failing

## Acceptance Criteria
- Improved test

### PR DESCRIPTION
**Pull request by DominicOram**
Fixes #653

Rather than check that the backlight has finished moving now we just check that we told it to move, which should be a lot less flaky.

To test:
* Run the unit tests all through a few times and confirm they do not fail with the backlight timeout error

### CODE REVIEW

File: tests/unit_tests/hyperion/experiment_plans/test_grid_detect_then_xray_centre_plan.py [hunk 1]
```diff
@@ -114,7 +114,9 @@ async def test_detect_grid_and_do_gridscan(
     mock_grid_detection_plan.assert_called_once()
 
     # Check backlight was moved OUT
-    assert await composite.backlight.position.get_value() == BacklightPosition.OUT
+    get_mock_put(composite.backlight.position).assert_called_once_with(
```
Comment (DiamondJoseph) [inline tests/unit_tests/hyperion/experiment_plans/test_grid_detect_then_xray_centre_plan.py:117]: Does your plan not wait on the move for the backlight? Or is the Sim backend not respecting where it does?

Comment (DominicOram) [reply tests/unit_tests/hyperion/experiment_plans/test_grid_detect_then_xray_centre_plan.py:117]: The plan goes on to call another plan that waits on the backlight. However, in this particular test we have mocked out that other plan. I guess I could have fixed this by not mocking out the other plan or having our mock do the wait but this seemed the simplest solution

**Review by dperl-dls — APPROVED** [review]
LGTM

File: tests/unit_tests/hyperion/experiment_plans/test_grid_detect_then_xray_centre_plan.py (hunk 1 above)
Comment (rtuck99) [reply tests/unit_tests/hyperion/experiment_plans/test_grid_detect_then_xray_centre_plan.py:123]: Hmm, this seems to be just as unreliable as the previous code

**Pull request merged**